source/jugglinglab/notation/HSS.java

```diff
@@ -0,0 +1,831 @@
+                    bncinfo.add(numBeats, new ArrayList<String>());
+    			    numBeats++;
+				    b1 = false;
+				    b2 = false;
+				    b3 = false;
+    			    continue;
+    		    } else if (Character.isWhitespace(c)) {    
+				    b1 = false;
+				    b2 = false;
+				    b3 = false;
+    			    continue;
+    		    } else if (c == 'B') {
+    		    	if (b1) {
+    		    		bncinfo.get(numBeats-1).set(subBeats,"B");
+    		    		b1 = false;
+    		    		b2 = true;
+    		    		continue;
+    		    	} else {
+    				    throw new JuggleExceptionUser("Syntax error in object pattern at position " + (i+1));
+    		    	}
+    		    } else if (c == 'F' || c == 'L') {
+    		    	if (b2) {
+    		    		bncinfo.get(numBeats-1).set(subBeats,"B" + Character.toString(c));
+    		    		b2 = false;
+    		    		continue;
+    		    	} else if (b3) {
+    		    		bncinfo.get(numBeats-1).set(subBeats,"BH" + Character.toString(c));
+    		    		b3 = false;
+    		    		continue;
+    		    	} else {
+    				    throw new JuggleExceptionUser("Syntax error in object pattern at position " + (i+1));
+    		    	}
+    		    } else if (c == 'H') {
+    		    	if (b2) {
+    		    		bncinfo.get(numBeats-1).set(subBeats,"BH");
+    		    		b2 = false;
+    		    		b3 = true;
+    		    	} else {
+    				    throw new JuggleExceptionUser("Syntax error in object pattern at position " + (i+1));
+    		    	}
+    		    } else {
+    			    throw new JuggleExceptionUser("Syntax error in object pattern at position " + (i+1));
+    		    }
+    	    } // if-else muxThrow
+        } //for
+        if (!muxThrow && minOneThrow ) {
+    	    if (throwSum%numBeats == 0) {
+    		    numObj = throwSum/numBeats;
+    	    } else {
+    	    	throw new JuggleExceptionUser("Bad average for object pattern");
+    	    }
+        } else {
+        	throw new JuggleExceptionUser("Syntax error in object pattern");
+        }
+        //append a space after setting the bounceinfo arraylist. Eventually bouncinfo
+        //is appended to iph so the space will get transferred there.
+        //Reason for space: 
+        //The primary reason is that if a multiplex throw involving a pass is created
+        //then the space will ensure the multiplex throw is written as, for example,
+        //[5p3 4] as opposed to [5p34]. In the former, a 5 is to be passed to juggler 3
+        //and simultaneously a multiplex 4 is to be thrown. In the latter, it'll get 
+        //interpreted as a 5 to be passed to juggler 34. In other cases, the space may 
+        //not matter. However, for coding convenience, the space is appended everywhere.
+        for (int i = 0; i < bncinfo.size(); i++) {
+        	for (int j = 0; j < bncinfo.get(i).size(); j++) {
+        		if (bncinfo.get(i).get(j) == "null") {
+        			bncinfo.get(i).set(j, "\s");
+        		} else {
+        			bncinfo.get(i).set(j, bncinfo.get(i).get(j) + "\s");
```

Comment:
Compiler error: illegal escape character
```suggestion
        			bncinfo.get(i).set(j, bncinfo.get(i).get(j) + " ");
```